x-pack/plugin/logsdb/src/yamlRestTest/resources/rest-api-spec/test/40_source_mode_setting.yml

```diff
@@ -0,0 +1,564 @@
+
+  # NOTE: the `index.mapping.source.mode` is ignored since time_series indices only supports synthetic source
+  - match: { test_time_series_index_mode_undefined.mappings._source.mode: synthetic }
+
+---
+create an index with logsdb index mode and stored source:
+  - do:
+      indices.create:
+        index: test_logsdb_index_mode_undefined
+        body:
+          settings:
+            index.mode: logsdb
+            index.mapping.source.mode: stored
+
+  - do:
+      indices.get_mapping:
+        index: test_logsdb_index_mode_undefined
+
+  # NOTE: the `index.mapping.source.mode` is ignored since time_series indices only supports synthetic source
+  - match: { test_logsdb_index_mode_undefined.mappings._source.mode: synthetic }
+
+---
+create an index with time_series index mode and disabled source:
+  - do:
+      indices.create:
+        index: test_time_series_index_mode
+        body:
+          settings:
+            index.mode: time_series
+            index.mapping.source.mode: disabled
+            routing_path: [ keyword ]
+          mappings:
+            properties:
+              keyword:
+                type: keyword
+                time_series_dimension: true
+
+  - do:
+      indices.get_mapping:
+        index: test_time_series_index_mode
+
+  # NOTE: the `index.mapping.source.mode` is ignored since time_series indices only supports synthetic source
+  - match: { test_time_series_index_mode.mappings._source.mode: synthetic }
+
+---
+create an index with logsdb index mode and disabled source:
+  - do:
+      indices.create:
+        index: test_logsdb_index_mode
+        body:
+          settings:
+            index.mode: logsdb
+            index.mapping.source.mode: disabled
+
+  - do:
+      indices.get_mapping:
+        index: test_logsdb_index_mode
+
+  # NOTE: the `index.mapping.source.mode` is ignored since logsdb indices only supports synthetic source
+  - match: { test_logsdb_index_mode.mappings._source.mode: synthetic }
+
+---
+modify final setting after index creation:
+  - do:
+      indices.create:
+        index: test_modify_setting
+        body:
+          settings:
+            index.mapping.source.mode: stored
+
+  - do:
+      catch: /.*Can't update non dynamic setting.*/
+      indices.put_settings:
+        index: test_modify_setting
+        body:
+          index.mapping.source.mode: synthetic
+
+---
+modify source mapping after index creation:
+  - do:
+      indices.create:
+        index: test_modify_source_mode
+        body:
+          settings:
+            index.mapping.source.mode: stored
+
+  - do:
+      catch: bad_request
+      indices.put_mapping:
+        index: test_modify_source_mode
+        body:
+          mappings:
+            _source: synthetic
+
+  - match: { error.type: "mapper_parsing_exception" }
+  - match: { error.reason: "Failed to parse mapping: Root mapping definition has unsupported parameters:  [mappings : {_source=synthetic}]" }
```

Comment:
I would expect a different error here? I thought we returned an error indicated that source mode can't be updated?

Reply:
Should this be:

```
- do:
      catch: bad_request
      indices.put_mapping:
        index: test_modify_source_mode
        body:
          mappings:
            _source: 
                mode: synthetic
```